nix/module.nix

```diff
@@ -0,0 +1,115 @@
+      type = bool;
+      default = false;
+      description = ''
+        Enable fw-fanctrl systemd service and install the needed packages.
+      '';
+    };
+    config = {
+      defaultStrategy = mkOption {
+        type = str;
+        default = defaultConfig.defaultStrategy;
+        description = "Default strategy to use";
+      };
+      strategyOnDischarging = mkOption {
+        type = str;
+        default = defaultConfig.strategyOnDischarging; 
+        description = "Default strategy on discharging";
+      };
+      batteryChargingStatusPath = mkOption {
+        type = str;
+        default = "/sys/class/power_supply/BAT1/status";
+        description = "";
+      };
+      strategies = mkOption {
+        default = defaultConfig.strategies;
+        type = attrsOf (submodule (
+          { options, name, ... }:
+          {
+            options = {
+              name = mkOption {
+                type = str;
+                default = "";
+                description = "Name of the strategy";
+              };
+              fanSpeedUpdateFrequency = mkOption {
+                type = int;
+                default = 5;
+                description = "How often the fan speed should be updated in seconds";
+              };
+              movingAverageInterval = mkOption {
+                type = int;
+                default = 25;
+                description = "Interval (seconds) of the last temperatures to use to calculate the average temperature";
+              };
+              speedCurve = mkOption {
+                default = [];
+                description = "How should the speed curve look like";
+                type = listOf (submodule (
+                  { options, ... }:
+                  {
+                    options = {
+                      temp = mkOption {
+                        type = int;
+                        default = 0;
+                        description = "Temperature at which the fan speed should be changed";
+                      };
+                      speed = mkOption {
+                        type = int;
+                        default = 0;
+                        description = "Percent how fast the fan should run at";
+                      };
+                    };
+                  }
+                ));
+              }; 
+            };
+          }
+        ));
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Install package
+    environment.systemPackages = [
+      fw-fanctrl
+      fw-ectool
+    ];
+
+    # Create config
+    environment.etc."fw-fanctrl/config.json" = {
+      text = builtins.toJSON cfg.config;
+    };
+
+    # Create Service
+    systemd.services.fw-fanctrl = {
+      description = "Framework Fan Controller";
+      after = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "simple";
+        Restart = "always";
+        ExecStart = "${fw-fanctrl}/bin/fw-fanctrl --run --config /etc/fw-fanctrl/config.json --no-log";
```

Comment:
I am against manually packaging the services.
(that was what the AUR package did in the beginning and it caused problems in the long run)

Reply:
I would love to just copy the `.service` file out of the repo in the right space, but that is not how NixOS works. I looked a lot if I could find a "converter" service file to nix or something like this, but I did not find anything. So we have to keep it like that.

Reply:
I understand, hopefully this can be fixed in the future.
The package being a separate branch, it should mitigate the problem a bit, we just have to be careful about changes to service files changes when updating the branch.